### ISSUE TITLE
What kind of tests should the template feature?

### ISSUE THREAD

**Issue by brianegan** [brianegan/new_flutter_template#8]
What kind of tests are useful in a template? Does it make sense to include tests at all? If so, what kind are helpful in a template?

- No Tests
- Placeholder files to add tests, but no actual tests in any of the files
- 1 Widget Test
- 1 Test per "Concept"  (e.g. if MVC pattern is used, include a test for the M, a test for the V, and a test for the C)
- Test for every file

**Comment by brianegan** [comment]
No Tests

**Comment by brianegan** [comment]
Placeholder files, but no tests

**Comment by brianegan** [comment]
1 Widget Test

**Comment by brianegan** [comment]
1 Test per "Concept" (e.g. if MVC pattern is used, include a test for the M, a test for the V, and a test for the C)

**Comment by brianegan** [comment]
Test for every file

**Comment by irvine5k** [comment]
Test every business logic and page.

**Comment by PiotrWpl** [comment]
I think that it would be a great idea to have three example tests:
* one unit test
* one widget test
* one integration test

this will allow new developers easily understand what is the difference between them in terms of the use-case and speed of executing.

**Comment by mhadaily** [comment]
Just to show how we can achieve 100% test coverage, can we test everything?  It may be helpful to see how to test some parts of the application that may be unclear how to test.

**Comment by burhanrashid52** [comment]
I won't look for 100% test coverage or test for every file. I agree with @PiotrWpl suggestion of having one test for each kind. It will help to understand the difference between all 3 and what to test and how much.

**Comment by brianegan** [comment]
Thanks everyone! @PiotrWpl very cool idea :)

@mhadaily Interesting idea! Do you think that would be best in the template itself or perhaps as part of a code lab / sample app of some kind? The only thing I worry about with adding 100% test coverage to a template is that folks will pretty much need to delete all of those tests right away. Therefore, it may be a nice demonstration that turns quickly annoying :P What do you think?

**Comment by mhadaily** [comment]
@brianegan  I agree with you, I believe it should be part of codelab or sample, but it's nice to show how we can achieve 100% for learning purposes.
